### ISSUE TITLE
Support for SSM Session Documents/Session Manager Preferences

### ISSUE THREAD

**Issue by gazoakley** [terraform-providers/terraform-provider-aws#5843]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

AWS have released SSM Session Manager, a tool that allows shell access to EC2 instances without the need to run SSH. Access is controlled through IAM permissions, and commands are logged.

Logging preferences are stored in an SSM document named SSM-SessionManagerRunShell which has a document type of "Session". This is a new document type, and as yet is unsupported by the Terraform AWS provider

### New or Affected Resource(s)

* aws_ssm_document

### Potential Terraform Configuration

```hcl
resource "aws_ssm_document" "session_manager_prefs" {
  name            = "SSM-SessionManagerRunShell"
  document_type   = "Session"
  document_format = "JSON"

  content = <<DOC
{
    "schemaVersion": "1.0",
    "description": "Document to hold regional settings for Session Manager",
    "sessionType": "Standard_Stream",
    "inputs": {
        "s3BucketName": "test",
        "s3KeyPrefix": "test",
        "s3EncryptionEnabled": true,
        "cloudWatchLogGroupName": "/logs/sessions",
        "cloudWatchEncryptionEnabled": false
    }
}
DOC
}
```

### References

* https://aws.amazon.com/blogs/aws/new-session-manager/
* https://docs.aws.amazon.com/systems-manager/latest/userguide/preference-setting-permissions.html

**Comment by gazoakley** [comment]
A prerequisite for this is updating to aws-sdk-go v1.15.33

**Comment by bflad** [comment]
Support for this has been merged into master and will release with version 1.36.0 of the AWS provider, likely later today. 👍

**Comment by bflad** [comment]
This has been released in [version 1.36.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1360-september-13-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Comment by gazoakley** [comment]
Thanks @bflad 🎉 

I've also published a module on the registry to allow setting session manager preferences: https://registry.terraform.io/modules/gazoakley/session-manager-settings
